### PR TITLE
sssd: Port to samba-4.12

### PR DESCRIPTION
**Pull request by gtgteq**
###### Motivation for this change
Fix `sssd`'s build. See https://github.com/SSSD/sssd/pull/976 and https://github.com/NixOS/nixpkgs/issues/82489

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/sssd/default.nix [hunk 1]
```diff
@@ -18,6 +18,8 @@ stdenv.mkDerivation rec {
     url = "https://fedorahosted.org/released/sssd/${pname}-${version}.tar.gz";
     sha256 = "0ngr7cgimyjc6flqkm7psxagp1m4jlzpqkn28pliifbmdg6i5ckb";
   };
+  # https://github.com/SSSD/sssd/pull/976
```
Comment (jtojnar) [inline pkgs/os-specific/linux/sssd/default.nix]: Reading the PR, it does not work and is buggy. Please use https://github.com/SSSD/sssd/commit/bc56b10aea999284458dcc293b54cf65288e325d via `fetchpatch` instead.

File: pkgs/development/libraries/tevent/default.nix [hunk 2]
```diff
@@ -20,7 +20,9 @@ stdenv.mkDerivation rec {
   '';
 
   configureFlags = [
-    "--bundled-libraries=NONE"
+    # "--bundled-libraries=NONE"
+    # ERROR: System library pytalloc-util of version 2.1.13 not found, and bundling disabled
+    "--bundled-libraries=pytalloc-util"
```
Comment (jtojnar) [inline pkgs/development/libraries/tevent/default.nix]: Please switch tevent to Python 3 as described in https://github.com/NixOS/nixpkgs/issues/82489.

**Review by jtojnar — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/tevent/default.nix (hunk 2 above)
Comment (gtgteq) [reply pkgs/development/libraries/tevent/default.nix]: Your PR is right and complete. I'll reduce this PR to contain only sssd's change.

File: pkgs/os-specific/linux/sssd/default.nix (hunk 1 above)
Comment (gtgteq) [reply pkgs/os-specific/linux/sssd/default.nix]: Done. I made change on the top of your PR. Is this right?

Comment (gtgteq) [reply pkgs/os-specific/linux/sssd/default.nix]: Wait. I didn't use `fetchpatch`. fixing it.

Comment (gtgteq) [reply pkgs/os-specific/linux/sssd/default.nix]: Done.

File: pkgs/os-specific/linux/sssd/default.nix [hunk 3]
```diff
@@ -18,6 +22,7 @@ stdenv.mkDerivation rec {
     url = "https://fedorahosted.org/released/sssd/${pname}-${version}.tar.gz";
     sha256 = "0ngr7cgimyjc6flqkm7psxagp1m4jlzpqkn28pliifbmdg6i5ckb";
   };
+  patches = [ patch-samba_4_12 ];
```
Comment (jtojnar) [inline pkgs/os-specific/linux/sssd/default.nix]: It is probably cleaner to put the `fetchpatch` invocation directly to `patches`. Also a comment would be nice. Example:

https://github.com/NixOS/nixpkgs/blob/46ec52f32966f16ce2bdd12bd150841e058988e9/pkgs/desktops/gnome-3/core/gnome-shell/default.nix#L83-L88

Comment (gtgteq) [reply pkgs/os-specific/linux/sssd/default.nix]: Done.

**Comment by jtojnar** [comment]
Thank you.

**Pull request merged**